lib/ecl/ecl_grid.cpp

```diff
@@ -7111,4 +7111,46 @@ void ecl_grid_export_index(const ecl_grid_type * grid, int * global_index, int *
       }
 }
 
+//This function is meant to be used w/ pandas datafram and numpy
+//Note: index_size must equal allocated size of output
+void ecl_grid_export_data_as_int( const ecl_grid_type * grid, int index_size, const int * data_index, ecl_kw_type * kw, int * output) {
+  int * input = ecl_kw_get_int_ptr(kw);
+  for (int i=0; i < index_size; i++) {
+    int di = data_index[i];
+    if (di >= 0)
+      output[i] = input[di];
+  }
+}
+
+//This function is meant to be used w/ pandas datafram and numpy
+//Note: index_size must equal allocated size of output
+void ecl_grid_export_data_as_double( const ecl_grid_type * grid, int index_size, const int * data_index, ecl_kw_type * kw, double * output) {
+  for (int i=0; i < index_size; i++) {
+    int di = data_index[i];
+    if (di >= 0)
+      output[i] = ecl_kw_iget_as_double(kw, di);
+  }
+}
+
+//This function is meant to be used w/ pandas datafram and numpy
+void ecl_grid_export_volume( const ecl_grid_type * grid, int index_size, const int * global_index, double * output ) {
+  for (int i = 0; i < index_size; i++) {
+    int g = global_index[i];
+    output[i] = ecl_grid_get_cell_volume1(grid, g);
+  }
+}
+
+//This function is meant to be used w/ pandas datafram and numpy
+void ecl_grid_export_position( const ecl_grid_type * grid, int index_size, const int * global_index, double * output) {
+  double x[1], y[1], z[1];
```

Comment:
That was a slightly weird construction; I would rather do:
```
double x,y,z;
...
...
ecl_grid_get_xyz1(grid, g, &x, &y, &z);
output[x];
```

alternatively you could completely drop the temporary variables `x,y`and `z` and just do:
```
ecl_grid_get_xyz(grid, g, &output[j], &output[j+1], &output[j+2])
```